provider/pkg/await/await.go

```diff
@@ -773,6 +776,10 @@ func Deletion(c DeleteConfig) error {
 				},
 				clientForResource: client,
 			})
+			if waitErr != nil {
```

Comment:
This is one bit here (and associated return) where it's not immediately obvious to a tired reader that control flow is not changing, and this is just a logging change, however I think that's still the case, it's not changing, localizing error handling to make the code better.. Appreciate eyes on this :)